code/modules/telescience/telescope/telescopeMisc.dm

```diff
@@ -264,3 +264,10 @@ var/list/magnet_locations = list()
 			new/obj/item/material_piece/iridiumalloy(src.loc)
 
 		..()
+
+	bullet_act(var/obj/projectile/P)
+		if (isobj(P.shooter))
+			var/obj/O = P.shooter
+			if(istype(O, /obj/critter/gunbot/drone/buzzdrone/naniteswarm)) //No more friendly fire
+				return
+		..()
```

Comment:
why not just fix them shooting from a distance

Reply:
Have now amended so saw drones & nanites attack from a distance of up to 1, every other drone attacks from a distance of up to 7 like they were previously.

Reply:
...and have reinstanted the bullet_act check: nanites were just piling on knocked down bodies and killing themselves again even with the range check.